### PR TITLE
refactor NearLimitRatio to environment variable

### PR DESCRIPTION
**Pull request by zufardhiyaulhaq**
Issue: https://github.com/envoyproxy/ratelimit/issues/159

According to this issue, I am moving NearLimitRatio constant 0.8 to a configurable environment variable with default is 0.8

### CODE REVIEW

**Comment by mattklein123** [comment]
Please check CI and also add documentation to README.md. Otherwise LGTM!

**Comment by zufardhiyaulhaq** [comment]
@mattklein123 I am not sure that this test fails because of my changes
```
--- FAIL: TestBasicAuthConfigWithRedisCluster (4.35s)
    --- FAIL: TestBasicAuthConfigWithRedisCluster/WithoutPerSecondRedisAuthWithLocalCache (1.09s)
        common.go:77: 
            	Error Trace:	common.go:77
            	            				integration_test.go:406
            	Error:      	Should be true
            	Test:       	TestBasicAuthConfigWithRedisCluster/WithoutPerSecondRedisAuthWithLocalCache
            	Messages:   	These two protobuf messages are not equal:
            	            	expected: overall_code:OK  statuses:{code:OK  current_limit:{requests_per_unit:20  unit:MINUTE}  limit_remaining:15  duration_until_reset:{seconds:19}}  statuses:{code:OK  current_limit:{requests_per_unit:10  unit:HOUR}  limit_remaining:5  duration_until_reset:{seconds:619}}
            	            	actual:  overall_code:OK  statuses:{code:OK  current_limit:{requests_per_unit:20  unit:MINUTE}  limit_remaining:15  duration_until_reset:{seconds:20}}  statuses:{code:OK  current_limit:{requests_per_unit:10  unit:HOUR}  limit_remaining:5  duration_until_reset:{seconds:620}}
```

I will add the documentation tomorrow. Thanks!

File: README.md [hunk 1]
```diff
@@ -412,7 +412,7 @@ STAT:
 * over_limit: Number of rule hits exceeding the threshold rate
 * total_hits: Number of rule hits in total
 
-These are examples of generated stats for some configured rate limit rules from the above examples:
+To costum near_limit ratio threshold, you can specify with `NEAR_LIMIT_RATIO` environment variable. It defaults to `0.8` (0-1 scale). These are examples of generated stats for some configured rate limit rules from the above examples:
```
Comment (mattklein123) [inline README.md:415]:
```suggestion
To use a custom near_limit ratio threshold, you can specify with `NEAR_LIMIT_RATIO` environment variable. It defaults to `0.8` (0-1 scale). These are examples of generated stats for some configured rate limit rules from the above examples:
```

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM with small comment, thanks.

/wait

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**